src/hooks/useExpandOnHover.ts

```diff
@@ -0,0 +1,78 @@
+import { useEffect, useRef, useState } from 'react'
+import { useSelector } from 'react-redux'
+import { expandBottom } from '../action-creators'
+import { DROP_TARGET } from '../constants'
+import { store } from '../store'
+import { Context, Path } from '../types'
+import { equalArrays, pathToContext } from '../util'
+import { State } from '../util/initialState'
+
+type Options = {
+  path?: Path,
+  depth: number,
+}
+
+/**
+ * Delays the truthy value by given time.
+ */
+const useDelayedTrigger = (trigger: boolean, triggerDelay: number) => {
+
+  const [value, setValue] = useState<boolean>(false)
+  const timeOutId = useRef<number | null>(null)
+
+  useEffect(() => {
+
+    if (trigger && !value) {
+      if (timeOutId.current) window.clearTimeout(timeOutId.current)
+      timeOutId.current = setTimeout(() => setValue(true), triggerDelay)
+    }
+    else if (!trigger && value && timeOutId.current) {
+      window.clearTimeout(timeOutId.current)
+      timeOutId.current = null
+      setValue(false)
+    }
+
+    return () => {
+      if (timeOutId.current) window.clearTimeout(timeOutId.current)
+    }
+  }, [trigger])
+
+  return value
+}
+
+/**
+ * Handles expansion of the context due to hover on the thought's empty drop.
+ */
+export const useExpandBottomOnHover = ({ path }: Options, expandDelay: number) => {
```

Comment:
The functionality looks good, just a couple questions:

1. `useExpandBottomOnHover` is tightly coupled to the `dragInProgress` reducer (via `state.hoveringPath` and `state.dragInProgress`) but this fact is quite hidden when you look at the hook and its arguments. I guess all hooks that use `useSelector` have that risk. Any ideas how to make these dependencies more explicit? We could pass them as arguments, but that just pushes `useSelector` to the component. Is it just a matter of clearer documentation that `useExpandBottomOnHover` only works with `dragInProgress`?

2. Could you clarify why this must be implemented as a hook instead of an action-creator? It looks like it just dispatches `expandBottom` based on `state.hoveringPath` and `state.dragInProgress`. Could this logic be called from the `dragInProgress` action-creator?

Reply:
> 2\. Could you clarify why this must be implemented as a hook instead of an action-creator? It looks like it just dispatches `expandBottom` based on `state.hoveringPath` and `state.dragInProgress`. Could this logic be called from the `dragInProgress` action-creator?

As you can see, `useExpandBottomOnHover` takes `path` as an argument.  The hook needs to maintain a timer for each thought. It would have been hard to maintain a timer for all thoughts with the action creator. That was the reason I implemented it as the hook. But I think it is possible to refactor this to use only a single timer just like with expand hover top. So I will change this to an action creator.